identity-apps-core/components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java

```diff
@@ -110,20 +110,32 @@ public static void createOAuth2Application(String applicationName, String portal
         oAuthConsumerAppDTO.setOauthConsumerKey(consumerKey);
         oAuthConsumerAppDTO.setOauthConsumerSecret(consumerSecret);
         String callbackUrl = IdentityUtil.getServerURL(portalPath, true, true);
-        if (!SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
-            callbackUrl = callbackUrl.replace(portalPath, "/t/" + tenantDomain.trim() + portalPath);
-        } else if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
-            if (StringUtils.equals(CONSOLE_APP, applicationName) &&
-                AppsCommonDataHolder.getInstance().isOrganizationManagementEnabled()) {
-                callbackUrl = "regexp=(" + callbackUrl + "|" + callbackUrl.replace(portalPath, "/t/(.*)" +
-                    portalPath) + "|" + callbackUrl.replace(portalPath, "/o/(.*)" + portalPath) + ")";
+        if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+            if (SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                if (StringUtils.equals(CONSOLE_APP, applicationName) &&
+                    AppsCommonDataHolder.getInstance().isOrganizationManagementEnabled()) {
+                    callbackUrl = "regexp=(" + callbackUrl
+                        + "|" + callbackUrl.replace(portalPath, "/t/(.*)" + portalPath)
+                        + "|" + callbackUrl.replace(portalPath, "/o/(.*)" + portalPath)
+                        + ")";
+                } else {
+                    callbackUrl = "regexp=(" + callbackUrl
+                        + "|" + callbackUrl.replace(portalPath, "/t/(.*)" + portalPath)
+                        + ")";
+                }
+            }
+        } else {
+            if (SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                callbackUrl = "regexp=(" + callbackUrl
+                    + "|" + callbackUrl.replace(portalPath, "/o/(.*)" + portalPath)
```

Comment:
why we need this 
`callbackUrl.replace(portalPath, "/o/(.*)" + portalPath)`
Do we access console as below.
`/o/<org-id>/console`

Reply:
We need to send logout requests with post redirect URL with a `/o/org_id` path for certain scenarios.